### PR TITLE
Added fully unboxed interface to Genotype.

### PR DESCRIPTION
**Pull request by cseed**
Use in expr.

### CODE REVIEW

**Comment by cseed** [comment]
Timing:

```
%%time
print(hc.read('profile225.vds')
      .filter_genotypes(
          '''
          g.dp > 400 ||
            (g.isHomRef && (g.ad[0] / g.dp < 0.9 || g.gq < 20)) ||
            (g.isHomVar && (g.ad[1] / g.dp < 0.9 || g.pl[0] < 20)) ||
            (g.isHet && ( (g.ad[0] + g.ad[1]) / g.dp < 0.9 || g.ad[1] / g.dp < 0.20 || g.pl[0] < 20 ))
          ''', keep=True)
 .count(genotypes=True))
```

master: 68s 66s 62s
this change: 57.1s 56.1s 60.3s

File: src/main/scala/is/hail/expr/FunctionRegistry.scala [hunk 1]
```diff
@@ -473,15 +459,50 @@ object FunctionRegistry {
...
+      null
+    else
+      box(gt)
+  })
+  registerMethod("gtj", { (x: Genotype) =>
+    val gt = x.unboxedGT
+    if (gt == -1)
+      null
+    else
+      Genotype.gtPair(gt).j: java.lang.Integer
```
Comment (tpoterba) [inline src/main/scala/is/hail/expr/FunctionRegistry.scala]: prefer consistency, either use box here or don't use box above

File: src/main/scala/is/hail/expr/FunctionRegistry.scala [hunk 2]
```diff
@@ -473,15 +459,50 @@ object FunctionRegistry {
...
+  })
+  registerMethod("gtk", { (x: Genotype) =>
+    val gt = x.unboxedGT
+    if (gt == -1)
+      null
+    else
+      Genotype.gtPair(gt).k: java.lang.Integer
```
Comment (tpoterba) [inline src/main/scala/is/hail/expr/FunctionRegistry.scala]: same

**Comment by cseed** [comment]
Addressed comments, back to you.

**Pull request merged**